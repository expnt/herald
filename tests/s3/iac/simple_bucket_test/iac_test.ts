import { testTerraform } from "../../../utils/iac.ts";
import * as path from "std/path/";
import { testDir } from "../../../utils/mod.ts";

Deno.test("Tofu resource provisioning - S3 backend", async (t) => {
  const tfDir = path.join(testDir, "s3/iac/simple_bucket_test");
  await testTerraform(tfDir, t);
});
