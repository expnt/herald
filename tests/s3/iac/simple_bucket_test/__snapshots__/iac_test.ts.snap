export const snapshot = {};

snapshot[`Tofu resource provisioning - S3 backend 1`] = `
\`
OpenTofu used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  [32m+[0m create[0m

OpenTofu will perform the following actions:

[1m  # aws_s3_bucket.example[0m will be created
[0m  [32m+[0m[0m resource "aws_s3_bucket" "example" {
      [32m+[0m[0m acceleration_status         = (known after apply)
      [32m+[0m[0m acl                         = (known after apply)
      [32m+[0m[0m arn                         = (known after apply)
      [32m+[0m[0m bucket                      = "s3-test"
      [32m+[0m[0m bucket_domain_name          = (known after apply)
      [32m+[0m[0m bucket_prefix               = (known after apply)
      [32m+[0m[0m bucket_regional_domain_name = (known after apply)
      [32m+[0m[0m force_destroy               = false
      [32m+[0m[0m hosted_zone_id              = (known after apply)
      [32m+[0m[0m id                          = (known after apply)
      [32m+[0m[0m object_lock_enabled         = (known after apply)
      [32m+[0m[0m policy                      = (known after apply)
      [32m+[0m[0m region                      = (known after apply)
      [32m+[0m[0m request_payer               = (known after apply)
      [32m+[0m[0m tags_all                    = (known after apply)
      [32m+[0m[0m website_domain              = (known after apply)
      [32m+[0m[0m website_endpoint            = (known after apply)
    }

[1m  # aws_s3_bucket_object.object[0m will be created
[0m  [32m+[0m[0m resource "aws_s3_bucket_object" "object" {
      [32m+[0m[0m acl                    = "private"
      [32m+[0m[0m arn                    = (known after apply)
      [32m+[0m[0m bucket                 = "s3-test"
      [32m+[0m[0m bucket_key_enabled     = (known after apply)
      [32m+[0m[0m content_type           = (known after apply)
      [32m+[0m[0m etag                   = "8898a737c3b4c5486d32ec2a31dbd86c"
      [32m+[0m[0m force_destroy          = false
      [32m+[0m[0m id                     = (known after apply)
      [32m+[0m[0m key                    = "new_object_key"
      [32m+[0m[0m kms_key_id             = (known after apply)
      [32m+[0m[0m server_side_encryption = (known after apply)
      [32m+[0m[0m source                 = "sample.txt"
      [32m+[0m[0m storage_class          = (known after apply)
      [32m+[0m[0m tags_all               = (known after apply)
      [32m+[0m[0m version_id             = (known after apply)
    }

[1mPlan:[0m 2 to add, 0 to change, 0 to destroy.
[0m[33mâ•·[0m[0m
[33mâ”‚[0m [0m[1m[33mWarning: [0m[0m[1mDeprecated Resource[0m
[33mâ”‚[0m [0m
[33mâ”‚[0m [0m[0m  with aws_s3_bucket_object.object,
[33mâ”‚[0m [0m  on main.tf line 7, in resource "aws_s3_bucket_object" "object":
[33mâ”‚[0m [0m   7: resource "aws_s3_bucket_object" "object" [4m{[0m[0m
[33mâ”‚[0m [0m
[33mâ”‚[0m [0muse the aws_s3_object resource instead
[33mâ”‚[0m [0m
[33mâ”‚[0m [0m(and one more similar warning elsewhere)
[33mâ•µ[0m[0m
[33mâ•·[0m[0m
[33mâ”‚[0m [0m[1m[33mWarning: [0m[0m[1mArgument is deprecated[0m
[33mâ”‚[0m [0m
[33mâ”‚[0m [0m[0m  with aws_s3_bucket_object.object,
[33mâ”‚[0m [0m  on main.tf line 8, in resource "aws_s3_bucket_object" "object":
[33mâ”‚[0m [0m   8:   bucket = [4maws_s3_bucket.example.bucket[0m[0m
[33mâ”‚[0m [0m
[33mâ”‚[0m [0mbucket is deprecated. Use the aws_s3_object resource instead.
[33mâ”‚[0m [0m
[33mâ”‚[0m [0m(and 2 more similar warnings elsewhere)
[33mâ•µ[0m[0m
[90m
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€[0m

Note: You didn't use the -out option to save this plan, so OpenTofu can't
guarantee to take exactly these actions if you run "tofu apply" now.
\`
`;
